hw2/task02/task02.py

```diff
@@ -0,0 +1,20 @@
+"""
+Given an array of size n, find the most common and the least common elements.
+
+The most common element is the element that appears more than n // 2 times.
+The least common element is the element that appears fewer than other.
+You may assume that the array is non-empty and the most common element
+always exist in the array.
+Example 1:
+Input: [3,2,3]
+Output: 3, 2
+Example 2:
+Input: [2,2,1,1,1,2,2]
+Output: 2, 1
+"""
+from collections import Counter
+
+
+def major_and_minor_elem(numbers: list) -> tuple:
+    counter = Counter(numbers)
+    return counter.most_common()[0][0], counter.most_common()[-1][0]
```

Comment:
What complexity these operations have?

Reply:
O(N) I assume @Cjkjvfnby

Reply:
Oh, I see the problem. Are there only 2 unique elements in any given array? I don't see any mention of it. @Cjkjvfnby

Reply:
> O(N) I assume  

Can you share more details on why do you assume it?

Reply:
the idea behind Counter is same as:
cache = {}
for item in numbers:
    if item in cache:
        cache[item] += 1
    else:
        cache[item] = 1

Reply:
`
```
cache = {}
for item in numbers:
    if item in cache:
        cache[item] += 1
    else:
        cache[item] = 1
```
`

Reply:
> the idea behind Counter is same as:

I want to emphasize that this comment is set on a specific line.  All lines before are O(N) but what about this line and the overall complexity of the function?

Reply:
You mean sorting of dict values that could be O(Nlog(N))? I noticed that Counter is coming out already sorted, so adressing .most_common by indexes is also O(N).
If not, this function could be rewritten using usual dict and manual search for min and max key-value pair

Reply:
> I noticed that Counter is coming out already sorted,

Can you be more specific, how exactly you have noticed that?  Which method do you use for it?

Reply:
In experimental manner, by printing it...

Reply:
At this point, you could just watch into the Counter implementation.

PS. There are 2 methods that responsible for printing the object. __str__ and __repr__.